libs/math/tests/unit/fixed_point/fixed_point.cpp

```diff
@@ -1146,58 +1174,56 @@ TYPED_TEST(TranscendentalTest, Exp)
   double    r;
   double    delta, max_error = 0, avg_error = 0;
   TypeParam x, scale;
-  scale               = 5.0;
+  scale               = fetch::math::AsType<TypeParam>(5.0);
   TypeParam tolerance = TypeParam::TOLERANCE;
   for (size_t i{0}; i < N; i++)
   {
     r                = static_cast<double>(std::rand()) / static_cast<double>(RAND_MAX);
-    x                = static_cast<TypeParam>(r) * (scale * 2) - scale;
+    x                = fetch::math::AsType<TypeParam>(r) * (scale * TypeParam{2}) - scale;
     TypeParam e      = TypeParam::Exp(x);
     double    e_real = std::exp(static_cast<double>(x));
     delta            = std::abs(static_cast<double>(e) - e_real);
     max_error        = std::max(max_error, delta);
     avg_error += delta;
   }
   avg_error /= static_cast<double>(N);
-  EXPECT_NEAR(max_error, 0.0, static_cast<double>(tolerance * 20));
+  EXPECT_NEAR(max_error, 0.0, static_cast<double>(tolerance) * 20);
   EXPECT_NEAR(avg_error, 0.0, static_cast<double>(tolerance));
 }
 
 TYPED_TEST(TranscendentalTest, Pow_positive_x_gt_1)
 {
-  TypeParam a{1.6519711627625};
+  TypeParam a{fetch::math::AsType<TypeParam>(1.6519711627625)};
   TypeParam two{2};
   TypeParam three{3};
-  TypeParam b{1.8464393615723};
-  TypeParam e1 = TypeParam::Pow(a, two);
-  TypeParam e2 = TypeParam::Pow(a, three);
-  TypeParam e3 = TypeParam::Pow(two, b);
+  TypeParam b{fetch::math::AsType<TypeParam>(1.8464393615723)};
+  TypeParam e1        = TypeParam::Pow(a, two);
+  TypeParam e2        = TypeParam::Pow(a, three);
+  TypeParam e3        = TypeParam::Pow(two, b);
+  auto      tolerance = static_cast<double>(TypeParam::TOLERANCE);
 
-  EXPECT_NEAR(static_cast<double>(e1 / std::pow(1.6519711627625, 2)), 1.0,
-              static_cast<double>(TypeParam::TOLERANCE));
-  EXPECT_NEAR(static_cast<double>(e2 / std::pow(1.6519711627625, 3)), 1.0,
-              static_cast<double>(TypeParam::TOLERANCE));
-  EXPECT_NEAR(static_cast<double>(e3 / std::pow(2, 1.8464393615723)), 1.0,
-              static_cast<double>(TypeParam::TOLERANCE));
+  EXPECT_NEAR(static_cast<double>(e1) / std::pow(1.6519711627625, 2), 1.0, tolerance);
+  EXPECT_NEAR(static_cast<double>(e2) / std::pow(1.6519711627625, 3), 1.0, tolerance);
+  EXPECT_NEAR(static_cast<double>(e3) / std::pow(2, 1.8464393615723), 1.0, tolerance);
 
   // Disabled for now, when fp128 will have full accuracy we will compare to the correct value
   // TypeParam base{1.0000018429229975};
   // TypeParam extreme{22741975.0640760175883770};
 
   double    r;
   double    delta, max_error = 0, avg_error = 0;
-  TypeParam x, y, scalex, scaley, margin, tolerance;
-  scalex    = 5.0;
-  margin    = 1.0;
-  tolerance = TypeParam::TOLERANCE;
+  TypeParam x, y;
+  double    scalex, scaley, margin;
```

Comment:
"double const" with immediate init would be great if you do not change the values.